src/Framework/MockObject/MockBuilder.php

```diff
@@ -178,6 +178,31 @@ public function setMethods(array $methods = null): self
         return $this;
     }
 
+    /**
+     * Specifies the subset of methods to mock, requiring each to exist in the class
+     */
+    public function setRealMethods(array $methods = null): self
+    {
+        if ($methods) {
+            $reflection = new \ReflectionClass($this->type);
```

Comment:
Argument 1 of ReflectionClass::__construct expects class-string<T>|object, parent type string provided
```
            $reflection = new \ReflectionClass($this->type);
                                               ^^^^^^^^^^^
```

Reply:
That doesn't seem to be accurate I don't think?

Reply:
There are a few ways to fix this.

1. change `if ($methods) {` to `if ($methods && class_exists($this->type)) {`
2. add `@psalm-var class-string` to the property type definition of `MockBuilder::$type`
3. add the error to the baseline with `vendor/bin/psalm --set-baseline=.psalm/baseline.xml`

Reply:
There's also a potential bug that @sebastianbergmann may have more insight into - the constructor allows a `string|array<string>`, but the property only allows `string`. I don't see how an `array` would work here.

Reply:
@muglug thanks a bunch for the help with that - I went with the 3rd approach for now, will see if there's any other feedback for that and please let me know if you see anything else.